.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+name: build
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+# Cancel any in-progress CI runs for a PR if it is updated
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    # The CMake configure and build commands are platform agnostic and should work equally well on Windows or Mac.
+    # You can convert this to a matrix build if you need cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-22.04 
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install dependencies
+      run: | 
+        sudo apt-get install \
+          libavfilter-dev \
+          libavformat-dev \
+          libavcodec-dev \
+          libswresample-dev \
+          libavutil-dev\
+          -y
+
+    - uses: Jimver/cuda-toolkit@v0.2.8
```

Comment:
not sure about this CI dependency. It seems to work for CUDA 11.7. Normal CUDA install instruction don't work. They use a special azure install script.

Reply:
this takes a long time and could make be cached

Reply:
looking at their CI, it is also much faster when only installing the needed components (nvcc, cuda runtime? is cuvid part of the driver or the CUDA Toolkit)